### ISSUE TITLE
Camera is broken

### ISSUE THREAD

**Issue by craigcabrey** [craigcabrey/thinkbook-13x-gen4-enablement#7]
6.10 adds an intree IPU6 driver, but userspace is still a mess. I managed to get it to the point of doing _something_, but still not video.

**Comment by wang1401** [comment]
That seem to be a camera through MIPI channel

**Comment by kmasterycsl** [comment]
So great. Looking forward to Fedora 41!

**Comment by croci** [comment]
Update: FOSS drivers still not ready but in the workings and hopefully will come soon:

https://hansdegoede.dreamwidth.org/28841.html

However, I just managed to make in-browser webcams work on Fedora 41 by installing the rpmfusion drivers:

https://hansdegoede.livejournal.com/27276.html

Note that you do not need to enable the updates-testing rpmfusion, the standard one works.

**Comment by craigcabrey** [comment]
I can confirm that the rpmfusion packages work on Fedora 41 now, only for browsers though. The quality isn't great, but it does work. I had to fiddle with `v4l2-relayd` configs to make it all work.

No wonder $INTC is in the dumpster.

**Comment by croci** [comment]
> I had to fiddle with v4l2-relayd configs to make it all work.

What do you mean by "to make it all work"? Did you manage to make the camera work outside of a browser as well? If so, can you share the workaround when you have time? Thanks!

**Comment by craigcabrey** [comment]
Seems to work out of the box on Fedora 42 now. I need to confirm with a clean install, but this has been an area of work in the Fedora community.

**Comment by bonjour81** [comment]
Hi !
Just wanted to share the following (for those of us that use other distro than Fedora) 
- I assume webcam sensor is OV02C10  based on output of dmesg | grep IPU6:
`intel-ipu6 0000:00:05.0: Found supported sensor OVTI02C1:00`
- Things are moving on this sensor:  https://lore.kernel.org/all/65a8dd5b-5737-4449-8ace-c325ed046884@linaro.org/T/

so hopefully some better Kernel support may show up soon 

Some Dell laptop: XPS9610, 9440 seems to use this sensor as well, so it may worth to keep an eye on discussions around those ones too

**Comment by bonjour81** [comment]
I patched my kernel with patch here:
https://patches.linaro.org/project/linux-media/patch/20250319145927.70534-1-hdegoede@redhat.com/

gnome camera now show a picture, upside down, with some glitch if moving, but that's a start!

**Comment by craigcabrey** [comment]
I managed to get the camera working in 41, with rpmfusion, some configs, and other packages. 

The 42 beta didn't have the camera working out of the box though.

**Comment by croci** [comment]
Used to work in Fedora 41 with rpmfusion. Now updated to 42 and camera is upside down. Great.

**Comment by drjhe** [comment]
Seems fixed with the latest ipu6 drivers from https://github.com/intel/ipu6-drivers (build from source). Need to poke the akmod maintainer to update the driver in rpmfusion.

**Comment by croci** [comment]
@drjhe Are there any updates on this? Did you poke the akmod maintainer?

I saw the discussion here https://github.com/intel/ipu6-drivers, but it mentions some libcamera bug and there is no related issue filed on the libcamera repo.

Thanks!

**Comment by croci** [comment]
Current state of things (very personal opinion ahead) is:

- The Intel IPU6 drivers are a mess: by the time they update them for a given kernel, the kernel gets updated by Fedora and it breaks the camera again.
- The open source drivers are getting there: Currently the camera works, but it is upside-down on F43 with kernel 6.17.11.  It seems like this upside-down issue will be fixed in Kernel 6.19 when it ships, see discussion [here](https://bugzilla.redhat.com/show_bug.cgi?id=2423139).

**Comment by drjhe** [comment]
> [@drjhe](https://github.com/drjhe) Are there any updates on this? Did you poke the akmod maintainer?
> 
> I saw the discussion here https://github.com/intel/ipu6-drivers, but it mentions some libcamera bug and there is no related issue filed on the libcamera repo.
> 
> Thanks!

Apologies for the very late reply. I believe the RPMFusion maintainer did apply an update. I've currently uninstalled the out-of-tree stuff, putting effort towards testing the in-tree stuff (whenever it arrives).

**Comment by bonjour81** [comment]
Camera seems ok now on 6.19.2-2, upside down solved (tested on CachyOS)

Does anyone heard anything related to IR camera ?

**Comment by croci** [comment]
Yes, it was fixed, see here: https://bugzilla.redhat.com/show_bug.cgi?id=2423139

Colors are still a bit off, but at least it seems to work and it is a solution that does not require using the intel ipu6-drivers which only officially support Ubuntu and are difficult to install and keep up to date on other distros.

**Comment by drjhe** [comment]
Camera's the right way up with kernel 6.19.8-200.fc43.x86_64, but I find I need to restart pipewire (in the user session) before the Camera app works properly.
